### PR TITLE
[Serve] Promote graceful shutdown and health check

### PR DESCRIPTION
**Pull request by sihanwang41**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by simon-mo — COMMENTED** [review]
can you please check the doc as well. For example,
```
doc/source/serve/deploying-serve.md:221:> @serve.deployment(_health_check_period_s=10, _health_check_timeout_s=30)
```

**Comment by sihanwang41** [comment]
> can you please check the doc as well. For example,
> 
> ```
> doc/source/serve/deploying-serve.md:221:> @serve.deployment(_health_check_period_s=10, _health_check_timeout_s=30)
> ```

done!

File: python/ray/serve/tests/test_schema.py [hunk 1]
```diff
@@ -478,66 +478,6 @@ def test_serve_application_invalid_import_path(self, path):
         with pytest.raises(ValidationError):
             ServeApplicationSchema.parse_obj(serve_application_schema)
 
-    def test_serve_application_aliasing(self):
```
Comment (simon-mo) [inline python/ray/serve/tests/test_schema.py:481]: @shrekris-anyscale anywhere else do we need to change?

**Review by simon-mo — APPROVED** [review]

File: python/ray/serve/tests/test_schema.py (hunk 1 above)
Comment (shrekris-anyscale) [reply python/ray/serve/tests/test_schema.py:481]: No, I believe everything is updated.

**Review by edoakes — COMMENTED** [review]
@sihanwang41 there is a note on the docs page, please remove it:
```

:::{note}
User-defined health checks are experimental and may be subject to change before the interface is stabilized. If you have any feedback or run into any issues or unexpected behaviors, please file an issue on GitHub.
:::
```

Then LGTM.

**Pull request merged**